plugins/mission/mission_impl.cpp

```diff
@@ -882,4 +884,107 @@ void MissionImpl::process_timeout()
+                                             const Json &qgc_plan_json)
+{
+    const auto json_mission_items = qgc_plan_json["mission"];
+
+    // Compose mission items by iterating JSON mission items
+    for (auto &json_mission_item : json_mission_items["items"].array_items()) {
+        // Parameters of Mission item & MAV command of it.
+        float speed_m_s = 0.f;
+        float gimbal_pitch_deg = 0.f, gimbal_yaw_deg = 0.f;
+        bool is_fly_through = false;
+        double lat_deg = NAN, lon_deg = NAN, rel_alt_deg = NAN;
+        double photo_inteval = 0.;
+        auto camera_action = MissionItem::CameraAction::NONE;
+        auto command = json_mission_item["command"].int_value();
+
+        // Extract parameters of each mission item
+        std::vector<double> params;
+        for (auto &p : json_mission_item["params"].array_items()) {
+            params.push_back(p.number_value());
+        }
+
+        switch (command) {
+            case MAV_CMD_IMAGE_START_CAPTURE:
+                photo_inteval = params[1];
+                camera_action = MissionItem::CameraAction::START_PHOTO_INTERVAL;
+                break;
+            case MAV_CMD_VIDEO_START_STREAMING:
+                camera_action = MissionItem::CameraAction::START_VIDEO;
+                break;
+            case MAV_CMD_VIDEO_STOP_STREAMING:
+                camera_action = MissionItem::CameraAction::STOP_VIDEO;
+                break;
+            case MAV_CMD_DO_CHANGE_SPEED:
+                speed_m_s = params[1];
+                break;
+            case MAV_CMD_DO_MOUNT_CONTROL:
+                gimbal_pitch_deg = -params[0]; // Pitch value is -ve as its in NED frame.
+                gimbal_yaw_deg = params[2];
+                break;
+            case MAV_CMD_NAV_WAYPOINT:
+                is_fly_through = (params[0] == 0.0) ? true : false;
+            case MAV_CMD_NAV_TAKEOFF:
+                lat_deg = params[4];
+                lon_deg = params[5];
+                rel_alt_deg = params[6];
+                break;
+            case MAV_CMD_NAV_RETURN_TO_LAUNCH:
+                break;
+            default: // Unsupported mission command
+                LogWarn() << "Mission: Unsupported mission command: " << command << ". Ignoring.";
+                break;
+        }
+        // Maybe removed
+        LogDebug() << "Mission: Cmd: " << command <<
+                   ", Lat: " << lat_deg << " deg, Lon: " << lon_deg << " deg, Rel alt: " << rel_alt_deg <<
+                   " meters, Speed: " << speed_m_s << " m/s, Is fly thru: " << (is_fly_through ? "true" : "false") <<
+                   " Gimbal pitch: " << gimbal_pitch_deg << " deg,  Gimbal yaw: " << gimbal_yaw_deg <<
+                   " deg, Camera action: " << MissionItem::to_str(camera_action) << "\n";
+
+        // Add mission item to the list
+        mission_items.push_back([&]() -> std::shared_ptr<MissionItem> const {
+            std::shared_ptr<MissionItem> new_item(new MissionItem());
```

Comment:
Instead of `std::shared_ptr<MissionItem> new_item(new MissionItem());` you can do `auto new_item = std::make_shared<MissionItem>();`.

Reply:
@julianoes I agree with you. How about adding Lets add it in [integration_tests/mission_qgc.cpp](https://github.com/dronecore/DroneCore/blob/develop/integration_tests/)

Reply:
It should be a unit test in my opinion.

Reply:
OK, you mean as part of [integration_tests/mission.cpp](https://github.com/dronecore/DroneCore/blob/develop/integration_tests/mission.cpp) ?

Reply:
No, not an integration test, a unit test.
https://github.com/dronecore/DroneCore/blob/develop/CMakeLists.txt#L221-L230

Reply:
Ok! Thanks. I'll add it.